src/hooks/useRange.js

```diff
@@ -0,0 +1,17 @@
+import { useState } from 'preact/hooks'
+
+export const useRange = (initialValue, range) => {
+  const [value, setValue] = useState(initialValue)
+
+  const [min, max] = Array.isArray(range) ? range : [0, range]
+  const setRangeValue = newValue => {
+    if (newValue < min) {
+      setValue(min)
+    } else if (newValue > max) {
+      setValue(max)
+    } else {
+      setValue(newValue)
+    }
+  }
+  return [value, setRangeValue]
```

Comment:
Would it make sense for `useRange` to expose `next()` and `previous()` functions?

Reply:
Good comment! It's totally make sense and that's the idea why another hook useRange here.